src/services/subject.js

```diff
@@ -1,12 +1,27 @@
+const mongoose = require('mongoose')
 const Subject = require('~/models/subject')
 const { createError } = require('../utils/errorsHelper')
 const { INTERNAL_SERVER_ERROR, INVALID_ID } = require('~/consts/errors')
 
 const subjectService = {
-  getSubjects: async () => {
+  getSubjects: async (match, sort, skip = 0, limit = 10) => {
     try {
-      const subjects = await Subject.find().exec()
-      return subjects
+      if (match.category) {
+        match.category = mongoose.Types.ObjectId(match.category)
```

Comment:
I think better to move this logic into model layer. Controller calls services, service calls Model and Model calls DataBase. In this case we can easily change our DB with updating only Model files.